src/go/rpk/pkg/cli/cmd/generate/grafana.go

```diff
@@ -411,7 +412,17 @@ func metricGroup(metric string) string {
 func fetchMetrics(
 	metricsEndpoint string,
 ) (map[string]*dto.MetricFamily, error) {
-	res, err := http.Get(metricsEndpoint)
+	req, err := http.NewRequestWithContext(
+		context.Background(),
+		http.MethodGet,
+		metricsEndpoint,
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	client := &http.Client{}
```

Comment:
This should be a package level variable with a timeout,
```go
var hClient = http.Client{Timeout: 10*time.Second}
```
A default client is even worse than the net/http one because a default client has no timeouts.

(also can we find a lint for `name := typeDefault{}`? it's weird to see `:= sync.Mutex{}` in a few places)